test/Calendar.test.js

```diff
@@ -6,34 +6,172 @@ const fixtures = require("./fixtures/index");
 
 const CALENDAR_DOMAIN = "http://example.com";
 const CALENDAR_PATH = "/cal.php/calendars/user/calendar_name/";
+const config = {
+    auth: {
+        user: "username",
+        pass: "password"
+    },
+    uri: CALENDAR_DOMAIN + CALENDAR_PATH
+};
 
 describe("Calendar", () => {
 
+    it("should throw if config object is not passed", () => {
+        const Calendar = createCalendar(() => {
+            return Promise.resolve(fixtures.response);
+        });
+
+        function calendar() {
+            return new Calendar();
+        }
+
+        expect(calendar).to.throw(Error, /Missing config object/);
```

Comment:
Use an arrow function here:

`expect(() => new Calendar())`;

Otherwise you should call the function rather `sandbox` than `calendar`.